psydac/fem/splines.py

```diff
@@ -239,10 +239,15 @@ def eval_field( self, field, *eta ):
         if self.basis == 'M':
             basis *= self._scaling_array[index]
 
-        return np.dot( field.coeffs[index], basis )
+        coeffs = field.coeffs[index].copy()
+
+        if weights:
+            coeffs *= wieghts[index]
```

Comment:
Typo: wieghts -> weights

I suspect that this code is never tested. Can you add a 1D unit test then?

Reply:
yes this part is never tested, how do we add a 1d test using `NurbsMapping`

Reply:
It should not be much different than a 2D test... You can create a 1D NURBS using Igakit

Reply:
we always return a TensorFemSpace when we use the `HDF5` files, so we will need to create everything by hand

Reply:
You're right, it does not really matter then.